TheMovieDb/Reviews/Controller/ReviewsCollectionViewController.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  ReviewsCollectionViewController.swift
+//  TheMovieDb
+//
+//  Created by Javier Cueto on 31/10/21.
+//
+
+import UIKit
+
+private let reuseIdentifier = "Cell"
+
+final class ReviewsCollectionViewController: UICollectionViewController {
+    
+    // MARK: - Properties
+    public var movie: Movie
+    private var reviews = [Review]()
+    
+    // MARK: - LifeCycle
+    init(with movie: Movie) {
+        self.movie = movie
+        let layout = UICollectionViewFlowLayout()
+        layout.sectionInset = .init(top: 0, left: 10, bottom: 0, right: 10)
+        super.init(collectionViewLayout: layout)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureUI()
+        fetchDataAPI()
+    }
+
+    // MARK: - Helpers
+    
+    private func configureUI() {
+        self.collectionView!.register(ReviewCell.self, forCellWithReuseIdentifier: ReviewCell.reuseIdentifier )
+        navigationItem.title = "Reviews"
+        collectionView.backgroundColor = .systemBackground
+    }
+    
+    // MARK: - API
+    
+    private func fetchDataAPI() {
+        let id = String(movie.id)
+        let parameter = APIParameters(id: id)
+        MovieAPI.shared.fetchData(endPoint: .review, with: parameter, completion: {(response: Result<Reviews, Error>) in
```

Comment:
If not wrong this could cause a retain cycle since a background call is made maybe a [weak self or unowned] (depending circunstance)

Reply:
In these case is not causing  retain cycle, deinit is reached when the controller is finished

Reply:
I know this may sound weird, but on the future someone could perform the request and then go to another view without waiting for response, but for now it's ok

Reply:
I gonna fix that in the next delivery